miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditSampleController.java

```diff
@@ -334,6 +393,245 @@ public JSONObject referenceDataJsonString() throws IOException {
 
     return hot;
   }
+  
+  @Autowired
+  private SampleClassService sampleClassService;
+  
+  @ModelAttribute("sampleClasses")
+  public List<SampleClass> getSampleClasses() throws IOException {
+    List<SampleClass> sampleClasses = new ArrayList<>(sampleClassService.getAll());
+    // Can't manually create Identities, so remove the option
+    for (Iterator<SampleClass> i = sampleClasses.iterator(); i.hasNext();) {
+      SampleClass sc = i.next();
+      if (Identity.CATEGORY_NAME.equals(sc.getSampleCategory())) {
+        i.remove();
+      }
+    }
+    Collections.sort(sampleClasses, new Comparator<SampleClass>() {
+      @Override
+      public int compare(SampleClass o1, SampleClass o2) {
+        return o1.getAlias().compareTo(o2.getAlias());
+      }
+    });
+    return sampleClasses;
+  }
+  
+  @Autowired
+  private TissueOriginService tissueOriginService;
+  
+  @ModelAttribute("tissueOrigins")
+  public List<TissueOrigin> getTissueOrigins() throws IOException {
+    List<TissueOrigin> list = new ArrayList<>(tissueOriginService.getAll());
```

Comment:
Yikes. We should introduce an `getAlias` and `getDescription` interfaces so most of this sorting can be done less painfully.